core/src/main/java/io/undertow/server/handlers/proxy/ProxyHandler.java

```diff
@@ -482,9 +482,13 @@ public void run() {
             final String remoteHost;
             final SocketAddress address = exchange.getSourceAddress();
             if (address != null) {
-                remoteHost = ((InetSocketAddress) address).getAddress().getHostAddress();
-                if(!((InetSocketAddress) address).isUnresolved()) {
-                    request.putAttachment(ProxiedRequestAttachments.REMOTE_ADDRESS, ((InetSocketAddress) address).getAddress().getHostAddress());
+                if (((InetSocketAddress) address).getAddress() != null){
+                    remoteHost = ((InetSocketAddress) address).getAddress().getHostAddress();
+                    if(!((InetSocketAddress) address).isUnresolved()) {
+                        request.putAttachment(ProxiedRequestAttachments.REMOTE_ADDRESS, ((InetSocketAddress) address).getAddress().getHostAddress());
```

Comment:
Should be:
request.putAttachment(ProxiedRequestAttachments.REMOTE_ADDRESS, remoteHost);

Reply:
Thanks @ropalka, fixed.